app/models/fee_scheme.rb

```diff
@@ -16,7 +16,7 @@ class FeeScheme < ApplicationRecord
     where('(:now BETWEEN start_date AND end_date) OR (start_date < :now AND end_date IS NULL)', now: Time.zone.now)
   }
   scope :for, lambda { |check_date|
-    where('(:date BETWEEN start_date AND end_date) OR (start_date < :date AND end_date IS NULL)', date: check_date)
+    where('(:date BETWEEN start_date AND end_date) OR (start_date <= :date AND end_date IS NULL)', date: check_date)
```

Comment:
👍 makes sense - could use coalesce on the between but I would not for now as have not tested.

```
scope :for, lambda { |check_date|
    where('(:date BETWEEN start_date AND COALESCE(end_date, now())', date: check_date)', date: check_date)
```